presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -520,6 +520,9 @@ private static String getColumnName(Session session, Metadata metadata, TableHan
 
                     if (indexExpression instanceof ConstantExpression) {
                         Object index = ((ConstantExpression) indexExpression).getValue();
+                        if (index == null) {
```

Comment:
Do we need the same check at L501?

Reply:
I haven't been able to come with a testcase where deference + index expression results in null though

Reply:
@bhhari I guess this is not possible. In that case, perhaps, add a verify(index != null, "<Fill in>") call.